frontend/src/views/Leaderboard.vue

```diff
@@ -49,6 +49,11 @@ export default {
   async created() {
     this.waitingLeaderboardOutcome = true;
     const leaderboardData = await fetch(apiUrl('static/leaderboard'));
+
+    if(leaderboardData.status === 429){
+      this.$dialog.notify.error('You are making too many requests. Please try again in 1 minute.');
```

Comment:
there should be a return here to bail as well

Reply:
Done in the new approach (delegate a success callback)